### PR TITLE
Move storage to common package

### PR DESCRIPTION
**Pull request by AndreiEres**

### CODE REVIEW

**Review by sandreim — COMMENTED** [review]
What is the reasoning for this change ? I would have expected this crate  to be named `storage` or just move out the `core` as a crate. The storage is a core functionality.

**Comment by AndreiEres** [comment]
> What is the reasoning for this change ? I would have expected this crate to be named `storage` or just move out the `core` as a crate. The storage is a core functionality.

Sorry I wasn't explicit. I am going to move all core as a package but I didn't want to do it in one time to prevent some hidden bugs. So I started with just a storage.

File: essentials/src/storage.rs [hunk 1]
```diff
@@ -164,6 +164,10 @@ pub trait RecordsStorage<K> {
 		K: Borrow<Q>;
 	/// Size of the storage
 	fn len(&self) -> usize;
+	/// Checks if the storage is empty
+	fn is_empty(&self) -> bool {
```
Comment (sandreim) [inline essentials/src/storage.rs:168]: This change seems to not be related to the PR.

**Review by sandreim — APPROVED** [review]
LGTM

**Pull request merged**